src/Rpc/NodeGuardService.cs

```diff
@@ -994,7 +996,45 @@ public override async Task<GetChannelResponse> GetChannel(GetChannelRequest requ
 
         return result;
     }
-    
+
+    public async Task<AddTagsResponse> AddTags(AddTagsRequest request, ServerCallContext context)
+    {
+        if (request.Tags.Count == 0)
+        {
+            throw new RpcException(new Status(StatusCode.InvalidArgument, "Tags are required"));
+        }
+        
+        foreach (var tag in request.Tags)
+        {
+            var utxo = tag.UtxoOutpoint.Split("-");
+            if (utxo.Length != 2)
+            {
+                throw new RpcException(new Status(StatusCode.InvalidArgument, $"Invalid outpoint {tag.UtxoOutpoint}"));
+            }
+
+            if (!uint.TryParse(utxo[1], out _))
+            {
+                throw new RpcException(new Status(StatusCode.InvalidArgument, $"Invalid output index {tag.UtxoOutpoint}"));
+            }
+        }
```

Comment:
you can validate the outpoint directly with `OutPoint.TryParse()`

Reply:
@RodriFS can you implement that?